Formula/otf2bdf.rb

```diff
@@ -33,7 +36,7 @@ class Otf2bdf < Formula
   end
 
   resource "mkinstalldirs" do
-    url "http://sofia.nmsu.edu/~mleisher/Software/otf2bdf/mkinstalldirs"
+    url "https://web.archive.org/web/20220629172211/http://sofia.nmsu.edu/~mleisher/Software/otf2bdf/mkinstalldirs"
```

Comment:
This file was named `sofia.nmsu.edu` instead of `mkinstalldirs`, so the `chmod 0755` on line 45 failed.

Reply:
There were two basenames (`sofia.nmsu.edu` and `web.archive.org`) that matched the `path.extname[query_regex]` [here](https://github.com/Homebrew/brew/blob/d87720333632459964ebce204fd33ecb377598d6/Library/Homebrew/download_strategy.rb#L361):

```
    # We need a Pathname because we've monkeypatched extname to support double
    # extensions (e.g. tar.gz).
    # Given a URL like https://example.com/download.php?file=foo-1.0.tar.gz
    # the basename we want is "foo-1.0.tar.gz", not "download.php".
    Pathname.new(uri_path).ascend do |path|
      ext = path.extname[query_regex]
      return path.basename.to_s[/#{query_regex.source}#{Regexp.escape(ext)}/] if ext
    end
```

And the first was returned:

```
$ brew irb
irb(main):001:0> url = "https://web.archive.org/web/20220629172211/http://sofia.nmsu.edu/~mleisher/Software/otf2bdf/mkinstalldirs"
=> "https://web.archive.org/web/20220629172211/http://sofia.nmsu.edu/~mleisher/Software/otf2bdf/mkinstalldirs"
irb(main):002:0> name = "mkinstalldirs"
=> "mkinstalldirs"
irb(main):003:0> AbstractFileDownloadStrategy.new(url, name, nil).instance_eval { parse_basename(url) }
=> "sofia.nmsu.edu"
```

Calling @Homebrew/brew for help.

Reply:
IMO it's not worth debugging this for a deprecated formula.

Reply:
> IMO it's not worth debugging this for a deprecated formula.

😅 True. To me it was more of a chance to familiarise myself with `Homebrew/brew`, I guess.

Reply:
@ZhongRuoyu No criticism here, don't worry, just hoping to save you some work ❤️